custom_components/pgnig_gas_sensor/PgpList.py

```diff
@@ -69,7 +69,10 @@ def from_dict(obj: Any) -> 'PpgListElement':
         assert isinstance(obj, dict)
         id_ppg = from_str(obj.get("IdPPG"))
         meter_number = from_str(obj.get("MeterNumber"))
-        reading_date_utc = from_datetime(obj.get("ReadingDateUtc"))
+        if obj.get("ReadingDateUtc") is not None:
+            reading_date_utc = from_datetime(obj.get("ReadingDateUtc"))
+        else:
+            reading_date_utc = datetime.utcnow()
```

Comment:
Well - this is not a proper solution to the problem.
_reading_date_utc_  is the date of the reading - the time when you or the technician added the meter reading to the system. This field is actually later on used to find the latest reading. If you fill it with utcnow() you may end up with an incorrect homeassistant state as the order may be broken.
It had a chance to work a few months ago as previously I was sorting the reading list by the value of reading, but it seems that some people are replacing the meter psychically and PGNIG is using the same ID for the new meter but the readings are starting from 0.

I hope this explains why I cannot accept the PR. What I would recommend is to create an issue and send me the jsons that you can have in your ebok. But please remember to review them and remove any personal data - or you can replace it with some fake stuff. What I am particularly interested in is which fields are null/empty which ones are filled and how they can be matched with each other. I believe that you know how to get this data as you did quite a nice MR.
Thank you for your contribution anyway.